vaporetto/src/sentence.rs

```diff
@@ -1203,6 +1209,31 @@ impl<'a, 'b> Token<'a, 'b> {
         &self.sentence.tags[start..end]
     }
 
+    /// Returns tag candidates with scores.
+    ///
+    /// The return value is a two-dimensional array. The outer array index corresponding to the
+    /// return value of [`Token::tags()`]. The inner array is a candidate set, where each element
+    /// is a tuple of the tag name and its score.
+    pub fn tag_candidates(&self) -> Vec<Vec<(&'b str, i32)>> {
+        let mut results = vec![];
+        if let Some((tags, scores)) = self.sentence.tag_scores[self.end - 1].as_ref() {
```

Comment:
If `Predictor::store_tag_scores` is unset, this line will panic implicitly due to "index out of bounds". Could you address it?

Reply:
@vbkaisetsu 

How about this suggestion?

More precisely, I mean that:

- `Token::tag_candidates` assumes that `self.sentence.tag_scores` is not empty.
- However, this instance is resized only when `Predictor::store_tag_scores` is set ([here](https://github.com/daac-tools/vaporetto/blob/tag-scores/vaporetto/src/predictor.rs#L562-L564)).
- If `Predictor::store_tag_scores` is unset, `Token::tag_candidates` will always panic implicitly because `self.sentence.tag_scores` is empty.

I think we should pre-check if `tag_scores` is empty or not and return None if empty.

Reply:
I added the Panics section in the previous commit.

I don't want to use Result here.
If the function returns `Result::Err`, it basically means that the conditions for normal completion have not been met at some point that cannot be controlled by the program, such as user input or external state.
This function should always return a normal value when `store_tag_scores()` is `true`, and I think it should panic instead of returning `Result` since this function should not be called when the flag is `false`.

Reply:
I overlooked the Panics section in the previous commit. Sorry.

I agree with your opinion.